overlord/configstate/config/transaction.go

```diff
@@ -361,3 +457,44 @@ func (e *NoOptionError) Error() string {
 	}
 	return fmt.Sprintf("snap %q has no %q configuration option", e.SnapName, e.Key)
 }
+
+// VirtualCfgFunc can be used for virtual "transaction.Get()" calls
+type VirtualCfgFunc func(snapName, key string, result VirtualResult) error
```

Comment:
why VirtualResult, can't this return an interface{} an the caller deals with the details?